### PR TITLE
267/display data in classrooms

### LINKED ISSUE #267: Import data in classrooms
We should be able to load : spaces, resources/actions for a given users

### PR DESCRIPTION
**Pull request by pyphilia**
close #267

### CODE REVIEW

**Comment by pyphilia** [comment]
This PR edit `StudentsTable` to display the imported data. We can now have spaces as additional columns, and actions and resources icons when they exist.

image

The edition functionalities are also improved to handle the data. 
When we want to edit a user, we can delete its related data:

image

When editing a classroom, we can remove spaces (and all their related data):
image

File: public/app/listeners/editClassroom.js [hunk 1]
```diff
@@ -15,9 +23,27 @@ const editClassroom = (mainWindow, db) => async (event, { name, id }) => {
...
+          .get(APP_INSTANCE_RESOURCES_COLLECTION)
+          .remove({ spaceId })
+          .write();
+      }
+    });
+
     // update data
     const now = new Date();
-    classroom.assign({ name, updatedAt: now }).write();
+    classroom.assign({ name, lastUpdatedAt: now }).write();
```
Comment (juancarlosfarah) [inline public/app/listeners/editClassroom.js:46]: I believe it should be `updatedAt`, right? We change it last time from this.

File: public/app/listeners/editUserInClassroom.js [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
-const { CLASSROOMS_COLLECTION, USERS_COLLECTION } = require('../db');
+const {
+  CLASSROOMS_COLLECTION,
+  USERS_COLLECTION,
+  ACTIONS_COLLECTION,
+  APP_INSTANCE_RESOURCES_COLLECTION,
+} = require('../db');
 const logger = require('../logger');
 
+// @param deleteSelection : map key-value with space id as key to whether the space data should be deleted
```
Comment (juancarlosfarah) [inline public/app/listeners/editUserInClassroom.js:11]: Should be in JSDoc comment format.

File: src/components/classrooms/EditClassroomButton.js [hunk 3]
```diff
@@ -1,74 +1,186 @@
...
+
+    // build delete selection on contained spaces for given user
+    const deleteSelection = buildDeleteSelection(spaces);
+
+    return {
+      open: false,
+      name,
+      deleteSelection,
+    };
+  })();
```
Comment (juancarlosfarah) [inline src/components/classrooms/EditClassroomButton.js:76]: Do you actually need to call the function with the last `();` bit? I don't remember it like that in the other classes.

File: src/components/classrooms/EditClassroomButton.js [hunk 4]
```diff
@@ -1,74 +1,186 @@
...
+      classes,
+      t,
+    } = this.props;
+    const { deleteSelection } = this.state;
+
+    return (
+      <>
+        <Typography className={classes.dataTitle}>{t('Spaces')}</Typography>
+
+        {spaces.length > 0 ? (
```
Comment (juancarlosfarah) [inline src/components/classrooms/EditClassroomButton.js:131]: `spaces.length` no need for `> 0`

**Comment by juancarlosfarah** [comment]
A small comment on the layout of the list of Actions and Resources itself, I think that the delete icon should be on the right. Also, can we use primary colours for icons? Or maybe a grayish one. These black ones are a bit harsh.

File: src/components/classrooms/ImportDataAssignUserForm.js [hunk 5]
```diff
@@ -135,24 +140,29 @@ class ImportDataAssignUserForm extends Component {
...
-      },
-    });
+    let username = null;
+    if (usernameOption) {
+      ({ value: username } = usernameOption);
+    }
+
+    // username cannot be null if actions or resources are imported
+    if (!username && (appInstanceResources || actions)) {
+      toastr.error(ERROR_MESSAGE_HEADER, 'A user need to be specified');
```
Comment (juancarlosfarah) [inline src/components/classrooms/ImportDataAssignUserForm.js:152]: `needs`. Do we factor out and translate?

File: src/utils/classroom.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+export const isClassroomNameValid = name => {
```
Comment (juancarlosfarah) [inline src/utils/classroom.js:1]: We might want to have simple, but specific unit tests for these functions.

File: src/utils/classroom.js [hunk 7]
```diff
@@ -1,6 +1,12 @@
 export const isClassroomNameValid = name => {
   // todo: check for special characters
-  return name.trim().length;
+
+  // check name is a string
+  if (typeof name !== 'string' && !(name instanceof String)) {
```
Comment (juancarlosfarah) [inline src/utils/classroom.js:5]: I would recommend using Lodash. https://lodash.com/docs#isString

**Pull request merged**